janus_core/calculations/geom_opt.py

```diff
@@ -223,17 +226,37 @@ def __init__(
             log_kwargs=log_kwargs,
             track_carbon=track_carbon,
             tracker_kwargs=tracker_kwargs,
+            file_prefix=file_prefix,
         )
 
         if not self.struct.calc:
             raise ValueError("Please attach a calculator to `struct`.")
 
-        # Set output file
+        # Set output files
         self.write_kwargs.setdefault("filename", None)
         self.write_kwargs["filename"] = self._build_filename(
             "opt.extxyz", filename=self.write_kwargs["filename"]
         ).absolute()
```

Comment:
I know this isn't you, probably me. Why set default, then immediately overwrite it?
```suggestion
        self.write_kwargs["filename"] = self._build_filename(
            "opt.extxyz", filename=self.write_kwargs.get("filename")
        ).absolute()
```

Reply:
Ah I think this was me, actually.

This construction was added in various places when we were allowing the file builder to create directories, whether or not the filename has been given, but I think it was just because I hadn't really thought about using `.get`.

Reply:
I haven't touched that in my edits yet, so that sounds good, cheers.